module/solutions/helper.js

```diff
@@ -1728,181 +2079,847 @@ module.exports = class SolutionsHelper {
+        //       userToken,
+        //       checkIfUserProjectExistsQuery,
+        //       ['_id'],
+        //     );
+        //     if (
+        //       checkForProjectExist.success &&
+        //       checkForProjectExist.data &&
+        //       checkForProjectExist.data.length > 0 &&
+        //       checkForProjectExist.data[0]._id != ''
+        //     ) {
+        //       checkForTargetedSolution.result['projectId'] = checkForProjectExist.data[0]._id;
+        //     }
+        //     // If project not found and createPrivateSolutionIfNotTargeted := true
+        //     // By default will be false for old version of app
+        //     if (!checkForTargetedSolution.result['projectId'] || checkForTargetedSolution.result['projectId'] === '') {
+        //       // user is not targeted and privateSolutionCreation required
+        //       /**
+        //        * function privateProgramAndSolutionDetails
+        //        * Request:
+        //        * @param {solutionData} solution data
+        //        * @param {userToken} for UserId
+        //        * @response private solutionId
+        //        */
+        //       let privateProgramAndSolutionDetails = await this.privateProgramAndSolutionDetails(
+        //         solutionData,
+        //         userId,
+        //         userToken,
+        //       );
+        //       if (!privateProgramAndSolutionDetails.success) {
+        //         throw {
+        //           status: httpStatusCode.bad_request.status,
+        //           message: messageConstants.apiResponses.SOLUTION_PROGRAMS_NOT_CREATED,
+        //         };
+        //       }
+        //       // Replace public solutionId with private solutionId.
+        //       if (privateProgramAndSolutionDetails.result != '') {
+        //         checkForTargetedSolution.result['solutionId'] = privateProgramAndSolutionDetails.result;
+        //       }
+        //     }
+        //   }
+        // }
+        delete checkForTargetedSolution.result['status'];
+        return resolve(checkForTargetedSolution);
       } catch (error) {
-        return reject(error);
+        return resolve({
+          success: false,
+          status: error.status ? error.status : httpStatusCode['internal_server_error'].status,
+          message: error.message,
+        });
       }
     });
   }
 
   /**
-   * Delete Solution.
+   * Verify Solution details.
    * @method
-   * @name deleteSolution
-   * @param {String} solutionId - solution Internal id.
-   * @param {String} userId - UserId.
-   * @returns {Object} Delete Solution .
+   * @name verifySolutionDetails
+   * @param {String} link - link Id.
+   * @param {String} userId - user Id.
+   * @param {String} userToken - user token.
+   * @returns {Object} - Details of the solution.
    */
 
-  static delete(solutionId = '', userId = '') {
+  static verifySolutionDetails(link = '', userId = '', userToken = '') {
```

Comment:
we were throwing error if user token in not present , Its defined in ml-core as well , But though it will be handled in auth middleware @VISHNUDAS-tunerlabs